### ISSUE TITLE
octavia-cli: import commands overwrite different resources with the same name

### ISSUE THREAD

**Issue by danieldiamond** [airbytehq/airbyte#14373]
## Expected behaviour
`octavia import ...` generates configuration files for a resource within a directory named `slugify(resource_name)`. however, if airbyte has multiple resources defined with the same name (different resource_id), octavia will ask to overwrite this config & state.

e.g.
```
resource_id,resource_name,directory_name
123,My Resource,my_resource
456,My Resource,my_resource
```

## Desired behaviour
Simply put, `octavia import...` should use an `get_output_path` function that relies on the resource ID and not the resource name. But ideally (in the interest of user-friendly directory names),`octavia import ...` will import resources in a way that is aware of the resource's ID and yet still write to some readable directory name 

One suggestion:
```
resource_id,resource_name,directory_name
123,My Resource,my_resource_123
456,My Resource,my_resource_456
```

**Comment by danieldiamond** [comment]
cc'ing @alafanechere

**Comment by danieldiamond** [comment]
Also noticing that `resource_id` isn't included in the configuration yaml. this makes sense if you're generating a template resource (the ID doesn't yet exist) but i think we should make resource_id optional in the renderer template and include it when writing out via import commands. thoughts?

**Comment by alafanechere** [comment]
> multiple resources defined with the same name (different resource_id), octavia will ask to overwrite this config & state.

This is a current limitation on the Airbyte back-end side: there is no uniqueness guarantee on resource names. I'd still prefer to enforce the unique resource names on the octavia side. 

> Simply put, octavia import... should use an get_output_path function that relies on the resource ID and not the resource name.

From my perspective the `import` should be run only once on a workspace and is not made to import the same resource from different workspace. E.G: You import your source from a workspace once, and then you can apply it on multiple workspace by switching the `--airbyte-url` or `--workspace-id`.  This is why I'm not prone to keep `resource_id` in configuration and would prefer to keep it in the files. I'd love to keep `configuration.yaml` workspace agnostic, especially for connections (see https://github.com/airbytehq/airbyte/issues/13748) .

Feel free to share your use case, I might not grasp something specific to your context.

**Comment by danieldiamond** [comment]
I like everything your saying unfortunately im referring to two resources with the same name. 

As an example, migrating a database with many large tables, there are many reasons why you would split this up into multiple source connectors e.g. to avoid resetting all tables when updating connection

Similarly, two source connectors to the same 3rd party application e.g. GitHub - i want want source connector to pull all repositories from all time. But I want another source connector to pull commits from a more recent start date (and a subset of repos). 

Each time you create another source connector for the same application, the default resource name is the same. 

So i guess we just need to make it clear to the user that every resource name (within a given resource type - source, destination, connection) needs to be unique. This will be annoying for users that have many different connections for a DB to Destination (one table per connection for a single source). They'd need to rename every resource. 
They should have this anyways, for readability/usability but it's not assumed.

**Comment by alafanechere** [comment]
Thank you for clarifying! I get the problem now. I'm on the side of waiting for an improvement on the platform side to force a unique resource name (I opened an issue [here](https://github.com/airbytehq/airbyte/issues/14448)). Users can still rename their resource on the console if they have an overwrite warning on `octavia import`

**Comment by danieldiamond** [comment]
👍 agreed, like it
